src/components/GasSelector.tsx

```diff
@@ -89,21 +106,28 @@ export default function GasSelector({
 
     try {
       const { network } = account;
-      const { fast } = await fetchGasPriceEstimates(network);
-      setGasPrice(fast.toString());
-      const fetchedGasPrice = hexWeiToString(inputGasPriceToHex(fast.toString()));
-      const fetchedNonce = await getNonce(network, account.address);
+      const [gas, fetchedNonce] = await Promise.all([
+        fetchUniversalGasPriceEstimate(network, account),
+        getNonce(network, account.address)
+      ]);
+      setGasPrice({
+        gasPrice: gas.gasPrice ?? '',
+        maxFeePerGas: gas.maxFeePerGas ?? '',
+        maxPriorityFeePerGas: gas.maxPriorityFeePerGas ?? ''
+      });
+      const txGas = mapObjIndexed((v) => v && inputGasPriceToHex(v), gas);
       setNonce(fetchedNonce.toString());
 
-      const txConfig: any = Object.assign({}, estimateGasCallProps, {
-        gasPrice: addHexPrefix(new BN(fetchedGasPrice).toString(16)),
-        nonce: fetchedNonce,
+      const txConfig: any = {
+        ...estimateGasCallProps,
+        ...txGas,
+        nonce: inputNonceToHex(fetchedNonce.toString()),
         chainId: network.chainId
-      });
+      };
       const fetchedGasLimit = await getGasEstimate(network, txConfig);
       setGasLimit(fetchedGasLimit);
     } catch (e) {
-      console.debug(e);
+      console.error(e);
```

Comment:
Can we test the error case so it never breaks anything ?

Reply:
What would you want tested? There isn't really a behaviour to test here 🤔

Reply:
Nevermind I misread it